feat: adds frontend specialist content (#13)
loft-br/hiring-process-study-guide: software-engineer/en_US/04.frontend-specialist.md

> # Frontend Specialist

Reviewer: Não gostei muito desse nome... que tal Frontend coding? Interface implementation? Ou qualquer outra coisa nessa vibe haha

Author: Eu também não curti muito, pois parece que tem a ver com o cargo 😬 
Frontend Skills? Que tal?

Reviewer: Gostei

> In this interview, we are going through a live coding, in which we are going to provide an online IDE with a boilerplate in the technology of your choice (pure javascript, react, Angular or Vue) so we can implement on screen and all logic related to it, based on low fidelity mockups.

Reviewer: ```suggestion
This interview consists on a live code session, for which we are going to provide an online IDE with a boilerplate in the technology of your choice (React, Angular, Vue or even vanilla Javascript) so we can implement a webpage and all the logic related to it, based on low fidelity mockups.
```

It is important to highlight that we won't run any of the written code nor check what is being rendered in the browser.
What we will do is write the code and talk about its implementation and possible evolutions.

> Our goal is to simulate a pair programming, close to what we do on a daily basis here at Loft.

Reviewer: ```suggestion
Our goal is to simulate a pair programming activity, close to what we do on a daily basis here at Loft.
```

We are going to present a task to be worked on that we will evolve as much as we can in the time we have available.

> You will be requested to share your screen so we can follow what is being developed.

Reviewer: ```suggestion
You will be requested to share your screen so we can watch what is being developed in real time.
```

Author: Concordo que follow não foi a melhor forma de falar; mas quis dizer no sentido de acompanhar mesmo, visto que a gente fala que é um pair programming e diz sobre construir em conjunto. Watch dá aquela impressão da pessoa ser observada e julgada sozinha, sabe? O que acha?

Reviewer: hmm talvez algo nessa linha:
```suggestion
You will be requested to share your screen so we can see your progress in real time and help you in a more timely manner.
```

Author: Show! ❤️

During the interview, we are going to ask questions to understand your decisions and to dig deeper on some topics of interest.

> You are the one that is going to guide us through this pairing and always feel free to tell us about what you are thinking, as well as any doubts that you might have along the way.

Reviewer: ```suggestion
You are the one that is going to guide us through this pairing. Please feel always free to tell us about what you are thinking, as well as any doubts that you might have along the way.
```

> Remember that there are no right or wrong answers, but many different ways to solve the same problema.  

Reviewer: ```suggestion
Remember that there are no right or wrong answers, but many different ways to solve the same problems.
```

> The most important for us is that we understand the thoughs behind your decisions!

Reviewer: ```suggestion
The most important for us is that we understand the reasoning behind your decisions!
```

## Main Evaluation Criteria
* Communication
* Solution Architecture and Evolution Vision
* How the solution was driven and concerns regarding quality and scale

> * CSS e interface implementation

Reviewer: ```suggestion
* CSS and interface implementation
```

* Readability
  * Code that is easy to read and understand
  * Proper separation of concerns
* Critical Sense

>   * Challenge interpretation and attention to requirements

Reviewer: Achei esquisita essa linha, mas não achei uma proposta melhor por ora 😬

Author: Challenge approach and attention to requirements? 
Challenge approach and making sure requirements are met?

Reviewer: Que tal:
```suggestion
  * Understanding the challenge
  * Meeting the requirements
```

Author: Eu gosto de usar `Understanding the challenge`, mas acho importante ressaltarmos a parte de atenção aos requisitos / entrega dos requisitos. Porque muita gente passa batido por muita coisa e muitos detalhes são importante para a gente, especialmente falando de Frontend. Acha que `Understanding the challenge` já cobre isso?

Reviewer: Foi o que eu tentei adicionando a segunda linha (Meeting the requirements) 😬
Mas pode seguir aí como achar melhor, acho que qualquer uma dessas vai transmitir bem a ideia

Author: Eu não tinha lido direito :moises:
Gostei assim 😄
